tests/ci/run_analytics.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+set -exo pipefail
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+source tests/ci/common_posix_setup.sh
+
+# The build and test time is not meant to be scientific benchmark, just to get a rough sense of changes overtime.
+# In case there is a weird change over time this recorded additional information about the host to the logs.
+lscpu
+
+branch=$(echo "$CODEBUILD_WEBHOOK_TRIGGER" | cut -d '/' -f2)
+common_dimensions="Branch=${branch}"
+
+function put_metric {
+  # This call to publish the metric could fail but we don't want to fail the build +e turns off exit on error
+  set +e
+  aws cloudwatch put-metric-data \
+    --namespace AWS-LC \
+    "$@" || echo "Publishing metric failed, continuing with the rest of the build"
+  # Turn it back on for the rest of the build
+  set -e
+}
+
+# Return the size of an object or total for the folder (summarize)
+function size {
+  du --bytes --apparent-size --summarize "$@" | cut -f1
+}
+
+SOURCE_CODE_SIZE=$(size "$SRC_ROOT")
+put_metric --metric-name FolderSize --value "$SOURCE_CODE_SIZE" --unit Bytes --dimensions "Folder=root,${common_dimensions}"
+
+for FOLDER_PATH in "$SRC_ROOT"/*/ ; do
+    FOLDER_SIZE=$(size "$FOLDER_PATH")
+    FOLDER=$(basename "$FOLDER_PATH")
+    put_metric --metric-name FolderSize --value "$FOLDER_SIZE" --unit Bytes --dimensions "Folder=${FOLDER},${common_dimensions}"
+done
+
+function run_build_and_collect_metrics {
```

Comment:
Nit: maybe we can move these metrics collection logics to `run_build` of `tests/ci/common_posix_setup.sh`.

Reply:
I thought about this but to maintain a more reasonable metrics history these builds are only triggered on push, not pr open/update. This is to prevent collecting weird data with PR's that add/remove a lot of code and then don't get merged in. Running this only on push makes sure it's what our branches actually looked like.